internal/statesync/reactor.go

```diff
@@ -158,8 +170,31 @@ func NewReactor(
 		blockStore:  blockStore,
 	}
 
+	if stateSync {
```

Comment:
I find this a possible source of confusion that you're constructing the state sync reactor with this state sync bool. I think it's fine for now, but it indicates to me that we should draw clear lines between running the service of providing state and running the syncing process itself.

Reply:
will remove it, reason same as above.

Reply:
so why the state sync reactor run the service but might not do the statesync things (If the statSync flag is false)?